config/nginx/django.conf

```diff
@@ -36,36 +33,43 @@ server {
 
   # Certbot configuration
   location /.well-known/acme-challenge/ {
-    root /var/www;
+      root /var/www;
   }
 
-  # Comment this if you enable HTTPS
   location / {
-        keepalive_timeout           500;
-        proxy_connect_timeout       600;
-        proxy_send_timeout          600;
-        send_timeout                600;
-        fastcgi_read_timeout        300;
-        proxy_read_timeout          600;
-        proxy_set_header Host            $host;
-        proxy_set_header X-Forwarded-For $remote_addr;
-        proxy_set_header        X-Forwarded-Proto $scheme;
-        proxy_pass http://web/;
+      keepalive_timeout                  500;
+      proxy_connect_timeout              600;
+      proxy_send_timeout                 600;
+      send_timeout                       600;
+      fastcgi_read_timeout               300;
+      # proxy_read_timeout                 600;
+      proxy_read_timeout                 120;
+      proxy_set_header Host              $host;
+      proxy_set_header X-Forwarded-For   $remote_addr;
+      proxy_set_header X-Forwarded-Proto $scheme;
+      proxy_pass http://web/;
   }
 
+  server_name $WEBGIS_PUBLIC_HOSTNAME;
+
+  # Listen
+  listen 8080;
+
   # !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
   # Uncomment the following lines if you want activate https
```

Comment:
@wlorenzetti from what I read [here](https://github.com/docker-library/docs/tree/master/nginx#using-environment-variables-in-nginx-configuration-new-in-119) it looks like that's feature already included in the docker version (BTW at the moment I don't have access to a real server and I can't tell how much of changes are correct..)

Reply:
@Raruto I just try for example to add new env variable: `$NGINX_CLIENT_MAX_BODY_SIZE`
I add it to `django.conf`:

```nginx
client_max_body_size $NGINX_CLIENT_MAX_BODY_SIZE;
```
and to `docker-compose.yml` file into `nginx` service:

```yml
...
    environment:
      - WEBGIS_PUBLIC_HOSTNAME
      - NGINX_CLIENT_MAX_BODY_SIZE
...      
```
But it not works, into nginx container log I found:
```
nginx: [emerg] "client_max_body_size" directive invalid value in /etc/nginx/conf.d/django.conf:1
```

As you can read [here](nginx: [emerg] "client_max_body_size" directive invalid value in /etc/nginx/conf.d/django.conf:1
) we have to create a nginx.conf.template file with environment variables, i.e. ${NGINX_CLIENT_MAX_BODY_SIZE} so that envsubst at startup can create the .conf file correctly.

Reply:
@wlorenzetti just a quick test to see how it could works: https://github.com/g3w-suite/g3w-suite-docker/pull/77/commits/8b3e51cee29e2ee5a9bfd45a391b4bcace606d88 (ref. to the `NGINX_CLIENT_MAX_BODY_SIZE` envsubst)

Probably, It won't work out of the box moving from g3w-suite v3.4 to v3.5 (having to edit the docker-compose.yml file as well), anyway, let me know if you see any other major issues.

For future reference, this is the code used by the docker image to make that substitution: [docker-nginx/entrypoint/20-envsubst-on-templates.sh](https://github.com/nginxinc/docker-nginx/blob/fef51235521d1cdf8b05d8cb1378a526d2abf421/entrypoint/20-envsubst-on-templates.sh)

Reply:
@Raruto I just pushed a commit to fix your changes: https://github.com/g3w-suite/g3w-suite-docker/pull/77/commits/36e6142b202e39c5d1881830e9e32f07bef29b79.
With your modifications `envsubst` builds the configuration file for nginx but removes the 'suffix', which in our case is` .conf` as seen from the entrypoint https://github.com/nginxinc/docker-nginx/ blob / fef51235521d1cdf8b05d8cb1378a526d2abf421 / entrypoint / 20-envsubst-on-templates.sh # L28.
It was enough to rename 'django.conf' to 'django.conf.template' and change NGINX_ENVSUBST_TEMPLATE_SUFFIX to `.template`